pkg/kubelet/status/generate.go

```diff
@@ -112,7 +113,7 @@ func GeneratePodReadyCondition(spec *v1.PodSpec, conditions []v1.PodCondition, c
 	// Generate message if any readiness gate is not satisfied.
 	unreadyMessages := []string{}
 	for _, rg := range spec.ReadinessGates {
-		_, c := podutil.GetPodConditionFromList(conditions, rg.ConditionType)
+		_, c := corev1helpers.GetPodConditionFromList(conditions, rg.ConditionType)
```

Comment:
It looks like `GeneratePodReadyCondition` could be changed to accept a `PodStatus` instead of 3 separate arguments. With that, we can make `GetPodConditionFromList` private.